plugins/discovery/telegraf/handler.go

```diff
@@ -61,6 +61,9 @@ func (e telegrafEncoder) Encode(ip, kind string, meta metav1.ObjectMeta, rule in
 	includeLabels := utils.Param(meta, discovery.LabelsAnnotation, cfg.IncludeLabels, "true")
 
 	values.Set("prefix", prefix)
+	values.Set("collectionInterval", cfg.Collection.Interval.String())
```

Comment:
should this be set only if non-zero?

Reply:
If is zero we will use the default value, so is not important.